### ISSUE TITLE
Treat Elvis operator as part of a call chain

### ISSUE THREAD

**Issue by sgrimm** [facebook/ktfmt#213]
Currently, it looks like an Elvis expression isn't counted as part of a call chain for purposes of figuring out where to put line breaks in the chain.

Example:

```kotlin
    val newWithdrawals =
        desiredWithdrawals?.filter {
          it.id == null && it.purpose != WithdrawalPurpose.GerminationTesting
        }
            ?: emptyList()
```

If I replace the Elvis operator with a chained method call, the format changes:

```kotlin
    val newWithdrawals =
        desiredWithdrawals
            ?.filter { it.id == null && it.purpose != WithdrawalPurpose.GerminationTesting }
            ?.also {}
```

and at that point, the Elvis operator gets formatted as part of the chain if I put it back:

```kotlin
    val newWithdrawals =
        desiredWithdrawals
            ?.filter { it.id == null && it.purpose != WithdrawalPurpose.GerminationTesting }
            ?.also {}
            ?: emptyList()
```

I think it would be more consistent if the first example were formatted like:

```kotlin
    val newWithdrawals =
        desiredWithdrawals
            ?.filter { it.id == null && it.purpose != WithdrawalPurpose.GerminationTesting }
            ?: emptyList()
```

Worth noting that this request possibly conflicts with #197.

**Comment by hick209** [comment]
With current version(after #416)
```
    val newWithdrawals =
        desiredWithdrawals?.filter {
          it.id == null && it.purpose != WithdrawalPurpose.GerminationTesting
        } ?: emptyList()
```
```
    val newWithdrawals =
        desiredWithdrawals
            ?.filter { it.id == null && it.purpose != WithdrawalPurpose.GerminationTesting }
            ?.also {} ?: emptyList()
```
```
    val newWithdrawals =
        desiredWithdrawals?.filter {
          it.id == null && it.purpose != WithdrawalPurpose.GerminationTesting
        } ?: emptyList()
```

Would that be an acceptable output for this case, @sgrimm?

**Comment by hick209** [comment]
Closing as I believe this is fixed. Feel free to reopen in case you think we still need to look into this

**Comment by sgrimm** [comment]
Oops, sorry, I missed the earlier comment here. The new behavior is definitely an improvement over the old one, but I think it would still be better to treat the operator as a step in the call chain. I'll file a new issue to clarify.
